wandb/sdk/interface/router.py

```diff
@@ -89,52 +70,20 @@ def message_loop(self) -> None:
             if self._mailbox:
                 self._mailbox.close()
 
-    def send_and_receive(
-        self,
-        rec: pb.Record,
-        local: bool | None = None,
-    ) -> MessageFuture:
-        rec.control.req_resp = True
-        if local:
-            rec.control.local = local
-        rec.uuid = uuid.uuid4().hex
-        future = MessageFutureObject()
-        with self._lock:
-            self._pending_reqs[rec.uuid] = future
-
-        self._send_message(rec)
-
-        return future
-
     def join(self) -> None:
         self._join_event.set()
         self._thread.join()
 
     def _handle_msg_rcv(self, msg: pb.Result | spb.ServerResponse) -> None:
-        if self._mailbox:
-            if isinstance(msg, pb.Result) and msg.control.mailbox_slot:
-                self._mailbox.deliver(
-                    spb.ServerResponse(
-                        request_id=msg.control.mailbox_slot,
-                        result_communicate=msg,
-                    )
-                )
-                return
-            elif isinstance(msg, spb.ServerResponse) and msg.request_id:
-                self._mailbox.deliver(msg)
-                return
-
-        if isinstance(msg, spb.ServerResponse):
-            msg = msg.result_communicate
-
-        with self._lock:
-            future = self._pending_reqs.pop(msg.uuid, None)
-        if future is None:
-            # TODO (cvp): saw this in tests, seemed benign enough to ignore, but
-            # could point to other issues.
-            if msg.uuid != "":
-                logger.warning(
-                    "No listener found for msg with uuid %s (%s)", msg.uuid, msg
-                )
+        if not self._mailbox:
             return
-        future._set_object(msg)
+
+        if isinstance(msg, pb.Result) and msg.control.mailbox_slot:
+            self._mailbox.deliver(
```

Comment:
why is this not covered?

Reply:
The only time `_read_message()` returns a `pb.Result` is in `RouterRelay`, but `RouterRelay` also overrides `_handle_msg_rcv()`. It has a call to `super()._handle_msg_rcv()` which I think never happens, but I am not sure enough to outright remove it.

Reply:
sounds good thanks!